rust/agents/relayer/src/relayer.rs

```diff
@@ -251,12 +264,35 @@ impl BaseAgent for Relayer {
 
         // send channels by destination chain
         let mut send_channels = HashMap::with_capacity(self.destination_chains.len());
-        for destination in &self.destination_chains {
+        for (dest_domain, dest_conf) in &self.destination_chains {
             let (send_channel, receive_channel) =
                 mpsc::unbounded_channel::<Box<DynPendingOperation>>();
-            send_channels.insert(destination.id(), send_channel);
+            send_channels.insert(dest_domain.id(), send_channel);
+
+            tasks.push(self.run_destination_submitter(dest_domain, receive_channel));
 
-            tasks.push(self.run_destination_submitter(destination, receive_channel));
+            let agent_metrics_conf = dest_conf
+                .agent_metrics_conf(Self::AGENT_NAME.to_string())
+                .await
+                .unwrap();
+            let agent_metrics_fetcher = dest_conf
+                .build_agent_metrics_fetcher(&self.core_metrics)
+                .await
+                .unwrap();
+            let agent_metrics = AgentMetrics::new(
+                self.agent_metrics.clone(),
+                agent_metrics_conf,
+                agent_metrics_fetcher,
+            );
+
+            let fetcher_task = tokio::spawn(async move {
+                agent_metrics
+                    .start_updating_on_interval(METRICS_SCRAPE_INTERVAL)
+                    .await;
+                Ok(())
+            })
+            .instrument(info_span!("AgentMetricsFetcher"));
```

Comment:
nit: maybe `"AgentMetrics"` as fetcher seems to be used to refer to the chain trait